python/src/equistore/tensor.py

```diff
@@ -88,6 +115,24 @@ def copy(self) -> "TensorMap":
         """
         return copy.deepcopy(self)
 
+    def __reduce_ex__(self, protocol: int):
+        """
+        Used by the Pickler to dump TensorMap object to bytes object.
+        When protocol >= 5 it supports PickleBuffer which reduces number of
+        copyies needed
+        """
+        path = os.path.join(tempfile.gettempdir(), str(id(self)))
+        try:
+            self._lib.eqs_tensormap_save(path.encode("utf8"), self._ptr)
```

Comment:
Same, this could use `equistore.io.save(path, self)`

Reply:
I think this was because io.py has a dependency on TensorMap, so I could not import it here